Angular points

edgargdev opened the pull request:
I need to do some refactoring of the authentication middleware. I just copied it over to the venues route file just so I could build out the backend points feature.

edgargdev committed:
key is being sent to server with venue name. NOTE: venue names must be unique at this point

edgargdev committed:
user is now able to get a point for correctly entered businessKey

edgargdev committed:
visited panel is now displaying users points/total # of venues

michaelachrisco commented on app/public/scripts/angular/panelStates/visited.js:
```diff
@@ -31,8 +35,8 @@ function VisitedController() {
       bar.text.style.fontSize = '2rem';
-
-      bar.animate(0.5);  // Number from 0.0 to 1.0
+      let percentage = user.points / venues.length
```
Nice!

michaelachrisco commented on app/backend/routes/venues.js:
```diff
@@ -10,4 +13,59 @@ router.get('/', (req,res,next) => {
+
 module.exports = router
+
+function ensureAuthenticated(req, res, next) {
```
I would add this check into the users model in `../models/user.js` or as a `../services/user.js` and use the method as a service.

michaelachrisco commented on app/backend/routes/venues.js:
```diff
@@ -10,4 +13,59 @@ router.get('/', (req,res,next) => {
   res.send(output)
 })
 
+router.post('/verify',ensureAuthenticated, (req, res, next) => {
```
At this point, do you mind refactoring this method into own controller and documented. This route file is doing a bit too much and is going to cause future confusion. I believe you alluded a bit to this.

michaelachrisco reviewed (changes requested):
Good PR. Just push the user route methods into the user model or a middleware/service and I think we should be good.

michaelachrisco commented:
As always @rxsharp  can override my opinion and merge, but I believe a small refactor needs to be done before this branch is merged. Not a huge deal but the route files are doing things the angular controllers (or middlewere depending) should be doing.

Thanks for the PR @Rawradin! Just make those small changes and Ill merge.

edgargdev committed:
refactor middleware into its own file as well as the user route/auth

edgargdev committed:
refactor some of the venues logic into a venues class

edgargdev commented:
Refactored most of the addressed points but could use a second opinion on how to proceed with all the User updating points logic in /venues/verify route.

edgargdev committed:
fixed visited panel bug. would happen on incorrect key submission

rick commented:
![](http://www.kickbright.net/photo2012/gbv5.gif)

[motor away](https://www.youtube.com/watch?v=9J-V6AGuA2k)

michaelachrisco commented:
@Rawradin LGTM!
